### PR TITLE
Optional misfit input

### PR DESCRIPTION
**Pull request by mloubout**
Add option to provide a per-shot misfit function to the FWI/LSRTM wrappers.

### CODE REVIEW

**Review by ziyiyin97 — COMMENTED** [review]
There are also a few typos in the `ipynb` file. If you need help on it let me know I can do a commit

File: src/TimeModeling/Modeling/losses.jl [hunk 1]
```diff
@@ -0,0 +1,15 @@
+export mse, studentst
+
+function mse(x, y)
```
[the remark this answers is not in the document]
Comment (mloubout) [reply src/TimeModeling/Modeling/losses.jl:3]: Has zero impact this is called from python that does a bunch of type conversions anyway.

File: examples/scripts/fwi_example_ADloss.jl [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+wavelet = ricker_wavelet(src_geometry.t[1],src_geometry.dt[1],0.008f0)  # 8 Hz wavelet
+q = judiVector(src_geometry,wavelet)
+
+############################### FWI ###########################################
+
+@everywhere function H(x)
+    n = size(x, 1)
+    σ = ifftshift(sign.(-n/2+1:n/2))
+    y = imag(ifft(σ.*fft(x, 1), 1))
+    return y
```
[the remark this answers is not in the document]
Comment (mloubout) [reply examples/scripts/fwi_example_ADloss.jl:40]: It's standard hilbert transform and in the notebook.

**Comment by mloubout** [comment]
@ziyiyin97 typos should be mostly fixed. Got PR lined up so will merge today unless something blocking

**Pull request merged**